File: Backend/Data.MyShop/Entities/BaseEntity.cs
using System.ComponentModel.DataAnnotations;

namespace Data.MyShop.Entities
{
    public interface IEntity<T>
    {
        T Id { get; set; }
        string Name { get; set; }
        bool IsDelete { get; set; }
        DateTime DateCreated { get; set; }
        DateTime DateUpdated { get; set; }
    }

    public abstract class BaseEntity<T> : IEntity<T>
    {
        [Key]
        public T Id { get; set; }
        public string Name { get; set; }
        public bool IsDelete { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public DateTime DateUpdated { get; set; } = DateTime.UtcNow;
    }
}

File: Backend/src/main/java/org/example/DAL/entities/BaseEntity.java
package org.example.DAL.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;
@Data //lombok {get; set;}
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@MappedSuperclass
public abstract class BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected Integer id;
    protected String name;
    protected boolean IsDelete = false;
    protected LocalDateTime dateCreated = LocalDateTime.now();
    protected LocalDateTime dateUpdated = LocalDateTime.now();
}
